api/src/main/java/com/okta/sdk/client/ClientBuilder.java

```diff
@@ -309,6 +315,45 @@
      */
     ClientBuilder setAuthenticationScheme(AuthenticationScheme authenticationScheme);
 
+    /**
+     * Allows specifying an authorization mode directly instead of relying on the
+     * default location + override/fallback behavior defined in the {@link ClientBuilder documentation above}.
+     *
+     * Authorization Mode could take one of values - SSWS (by default), PrivateKey (for OAUTH2).
+     *
+     * @param authorizationMode mode of authorization for requests to the Okta API server.
+     * @return the ClientBuilder instance for method chaining.
+     */
+    ClientBuilder setAuthorizationMode(String authorizationMode);
+
+    /**
+     * Allows specifying a list of scopes directly instead of relying on the
+     * default location + override/fallback behavior defined in the {@link ClientBuilder documentation above}.
+     *
+     * @param scopes list of scopes for which the client requests access.
+     * @return the ClientBuilder instance for method chaining.
+     */
+    ClientBuilder setScopes(List<String> scopes);
+
+    /**
+     * Allows specifying the private key (PEM file) path (for private key jwt authentication) directly instead
```

Comment:
key file or armored PEM string?

Reply:
path to private key pem file.

Reply:
@bretterer I assume both are needed for the first version to support 12-factor (all of the other config properties are)?

Reply:
@bdemers raised this question (whether we need to support 'path to private key' or 'armored string' or 'both'). I got answer that its upto us to decide. I have implemented code to support the 'path to private key pem file' way.